homeassistant/components/starline/__init__.py

```diff
@@ -46,10 +50,18 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     async def async_set_scan_interval(call):
         """Service for set scan interval."""
         options = dict(config_entry.options)
-        options[CONF_SCAN_INTERVAL] = call.data[CONF_SCAN_INTERVAL]
+        if call.service == SERVICE_SET_SCAN_INTERVAL:
+            options[CONF_SCAN_INTERVAL] = call.data[CONF_SCAN_INTERVAL]
+        elif call.service == SERVICE_SET_SCAN_OBD_INTERVAL:
+            options[CONF_SCAN_OBD_INTERVAL] = call.data[CONF_SCAN_INTERVAL]
         hass.config_entries.async_update_entry(entry=config_entry, options=options)
 
-    hass.services.async_register(DOMAIN, SERVICE_UPDATE_STATE, account.update)
+    async def async_update(call=None):
+        """Service for update all data."""
```

Comment:
Is it normal here to use this kind of function documentation when the method is called by a service? Normally the documentation would be something like "Update all data.", right? Or, since this function is defined inside another function, does it need to have such documentation? It matches the other function above, at least.